### PR TITLE
#295 Remove Navigation Drawer

### LINKED ISSUE #295: Remove Navigation Drawer
Keep:
- Navigation Bar in portrait
- Navigation Rail in landscape

### PR DESCRIPTION
**Pull request by bartek977**
Closes #295

### CODE REVIEW

File: app/src/main/kotlin/com/intive/picover/main/navigation/model/NavigationType.kt [hunk 1]
```diff
@@ -6,15 +6,13 @@ import androidx.compose.material3.windowsizeclass.WindowWidthSizeClass
...
 	;
 
 	companion object {
 		fun fromWindowSize(windowSize: WindowSizeClass) =
 			when (windowSize.widthSizeClass) {
-				WindowWidthSizeClass.Medium -> NAVIGATION_RAIL
-				WindowWidthSizeClass.Expanded -> NAVIGATION_DRAWER
-				else -> NAVIGATION_BAR
+				WindowWidthSizeClass.Compact -> NAVIGATION_BAR
+				else -> NAVIGATION_RAIL
```
Comment (MarekMacko) [inline app/src/main/kotlin/com/intive/picover/main/navigation/model/NavigationType.kt]: 1. use if-else
2. Please also simplify names, there is no need to repeat "NAVIGATION_" in every enum value

Comment (bartek977) [reply app/src/main/kotlin/com/intive/picover/main/navigation/model/NavigationType.kt]: Done

Comment (MarekMacko) [reply app/src/main/kotlin/com/intive/picover/main/navigation/model/NavigationType.kt]: Please check where I posted a comment. I was referring to `NavigationType` class and `fromWindowSize` function.

File: app/src/main/kotlin/com/intive/picover/main/view/MainActivity.kt [hunk 2]
```diff
@@ -75,54 +74,55 @@ class MainActivity : ComponentActivity() {
...
-				PicoverNavigationDrawer(
-					items = navigationItems,
-					navController = navController,
-					onItemClick = {
-						navController.navigateWithSingleTop(it)
-					},
-					modifier = Modifier.padding(it),
-				)
+		Scaffold { paddingValues ->
+			if (navigationType == NavigationType.BOTTOM_BAR) {
```
Comment (MarekMacko) [inline app/src/main/kotlin/com/intive/picover/main/view/MainActivity.kt]: Here you should use `when` to due to its exhaustive feature

File: app/src/main/kotlin/com/intive/picover/main/view/MainActivity.kt [hunk 3]
```diff
@@ -75,54 +74,55 @@ class MainActivity : ComponentActivity() {
...
+				ContentWithNavigationBar(navigationItems, navController, paddingValues)
+			} else {
+				ContentWithNavigationRail(navigationItems, navController, paddingValues)
 			}
 		}
 	}
 
+	@Composable
+	fun ContentWithNavigationBar(
```
Comment (MarekMacko) [inline app/src/main/kotlin/com/intive/picover/main/view/MainActivity.kt]: Add `private`

File: app/src/main/kotlin/com/intive/picover/main/view/MainActivity.kt [hunk 4]
```diff
@@ -75,54 +74,55 @@ class MainActivity : ComponentActivity() {
...
+					navController.navigateWithSingleTop(it)
+				}
+			},
+			modifier = Modifier.padding(paddingValues),
+			drawerState = drawerState,
+		)
+	}
+
+	@Composable
+	fun ContentWithNavigationRail(
```
Comment (MarekMacko) [inline app/src/main/kotlin/com/intive/picover/main/view/MainActivity.kt]: Add `private`

**Comment by MarekMacko** [comment]
@bartek977 Please add comment to the description to enable issue auto-close

File: app/src/main/kotlin/com/intive/picover/main/navigation/model/NavigationType.kt (hunk 1 above)
Comment (bartek977) [reply app/src/main/kotlin/com/intive/picover/main/navigation/model/NavigationType.kt]: Fixed :D

File: app/src/main/kotlin/com/intive/picover/main/view/MainActivity.kt (hunk 2 above)
Comment (bartek977) [reply app/src/main/kotlin/com/intive/picover/main/view/MainActivity.kt]: Done

File: app/src/main/kotlin/com/intive/picover/main/view/MainActivity.kt (hunk 3 above)
Comment (bartek977) [reply app/src/main/kotlin/com/intive/picover/main/view/MainActivity.kt]: Done

File: app/src/main/kotlin/com/intive/picover/main/view/MainActivity.kt (hunk 4 above)
Comment (bartek977) [reply app/src/main/kotlin/com/intive/picover/main/view/MainActivity.kt]: Done

**Comment by bartek977** [comment]
> @bartek977 Please add comment to the description to enable issue auto-close

Done

**Review by MarekMacko — APPROVED** [review]

**Review by michal-kucznerowicz — APPROVED** [review]

**Pull request merged**